components/GroupLinkSearch.js

```diff
@@ -0,0 +1,60 @@
+import { useState } from "react";
+import { Combobox } from "@headlessui/react";
+import Label from "./form/Label";
+
+function GroupLinkSearch({ selectedGroup, handleGroupSelection, groups }) {
+  const [query, setQuery] = useState("");
+
+  const handleGroupChange = (e) => {
```

Comment:
I find _e_ a bit ambiguous, consider renaming it to `event`. This can enhance code readability and understanding.
Renaming it to `event` makes it clear that this an _event handler_.

Reply:
I think we use `e` in other places, so should not be an issue, but thank you for highlighting